### PR TITLE
Fix rendering test module

### PR DESCRIPTION
**Pull request by jianhuang01**
#999 initial work to make the rendering test utility code work.

### CODE REVIEW

File: src/core/pure-js/deck-js.js [hunk 1]
```diff
@@ -240,7 +240,8 @@ export default class DeckGLJS {
       // Helps debug layer picking, especially in framebuffer powered layers
       drawPickingColors: this.props.drawPickingColors
     });
-    this.props.onAfterRender({gl}); // TODO - should be called by AnimationLoop
+    // TODO - should be called by AnimationLoop
+    this.props.onAfterRender({gl, canvas: this.props.canvas});
```
Comment (ibgreen) [inline src/core/pure-js/deck-js.js]: You can use [`gl.canvas`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/canvas)

File: test/rendering-test/test-config.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ import {
 
 } from 'deck.gl';
 
-export const WIDTH = 1600;
-export const HEIGHT = 900;
+export const WIDTH = 800;
```
Comment (ibgreen) [inline test/rendering-test/test-config.js]: There is some downscaling support in luma.gl if we want to render in high resolution but show smaller images

File: test/rendering-test/test-rendering.js [hunk 3]
```diff
@@ -113,7 +113,7 @@ class RenderingTest extends Component {
     });
   }
 
-  _onDrawComplete(name, referecenResult, canvas) {
+  _onDrawComplete(name, referecenResult, callbackData) {
```
Comment (ibgreen) [inline test/rendering-test/test-rendering.js]: Nit: We usually call these something like simple like `opts`. `callbackData` is more C style. But maybe a good idea...

Nit: `referecenResult` is misspelled

File: test/rendering-test/test-rendering.js [hunk 4]
```diff
@@ -155,7 +155,11 @@ class RenderingTest extends Component {
       height: height,
       debug: true,
       onAfterRender: this._onDrawComplete.bind(this, name, referecenResult)
-    }, mapViewState, {
+    },
+    {
+      viewports: [new WebMercatorViewport(viewportProps)]
```
Comment (Pessimistress) [inline test/rendering-test/test-rendering.js]: Why use `_extends` instead of a plain object?

Comment (Pessimistress) [reply test/rendering-test/test-rendering.js]: Also just use `viewport`. `viewports` is still experimental.

File: src/core/pure-js/deck-js.js (hunk 1 above)
Comment (jianhuang01) [reply src/core/pure-js/deck-js.js]: Thanks for the suggestion, fixed

File: test/rendering-test/test-rendering.js (hunk 3 above)
Comment (jianhuang01) [reply test/rendering-test/test-rendering.js]: thanks.

File: test/rendering-test/test-rendering.js (hunk 4 above)
Comment (jianhuang01) [reply test/rendering-test/test-rendering.js]: fixed

**Pull request merged**